okta/resource_okta_app_oauth_post_logout_redirect_uri.go

```diff
@@ -0,0 +1,105 @@
+package okta
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/okta/okta-sdk-golang/v2/okta"
+)
+
+func resourceAppOAuthPostLogoutRedirectURI() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAppOAuthPostLogoutRedirectURICreate,
+		ReadContext:   resourceAppOAuthPostLogoutRedirectURIRead,
+		UpdateContext: resourceAppOAuthPostLogoutRedirectURIUpdate,
+		DeleteContext: resourceAppOAuthPostLogoutRedirectURIDelete,
+		// The id for this is the uri
+		Importer: createCustomNestedResourceImporter([]string{"app_id", "id"}, "Expecting the following format: <app_id>/<uri>"),
+		Schema: map[string]*schema.Schema{
+			"app_id": {
+				Required: true,
+				Type:     schema.TypeString,
+				ForceNew: true,
+			},
+			"uri": {
+				Required:         true,
+				Type:             schema.TypeString,
+				Description:      "Post Logout Redirect URI to append to Okta OIDC application.",
+				ValidateDiagFunc: stringIsURL(validURLSchemes...),
+			},
+		},
+	}
+}
+
+func resourceAppOAuthPostLogoutRedirectURICreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	err := appendPostLogoutRedirectURI(ctx, d, m)
+	if err != nil {
+		return diag.Errorf("failed to create post logout redirect URI: %v", err)
+	}
+	d.SetId(d.Get("uri").(string))
+	return resourceAppOAuthPostLogoutRedirectURIRead(ctx, d, m)
+}
+
+// read does nothing due to the nature of this resource
+func resourceAppOAuthPostLogoutRedirectURIRead(context.Context, *schema.ResourceData, interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceAppOAuthPostLogoutRedirectURIUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	if err := appendPostLogoutRedirectURI(ctx, d, m); err != nil {
+		return diag.Errorf("failed to update post logout redirect URI: %v", err)
+	}
+	// Normally not advisable, but ForceNew generated unnecessary calls
+	d.SetId(d.Get("uri").(string))
+	return resourceAppOAuthPostLogoutRedirectURIRead(ctx, d, m)
+}
+
+func resourceAppOAuthPostLogoutRedirectURIDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	appID := d.Get("app_id").(string)
+
+	oktaMutexKV.Lock(appID)
```

Comment:
@jmaness is your PR complete? I'm not seeing were oktaMutexKV is initialized with a call to the kv mutex constructor `func NewMutexKV()`

Reply:
`oktaMutexKV` is initialized in `provider.go` at https://github.com/okta/terraform-provider-okta/pull/931/files#diff-a44dd20b4a48cf4cb52debcb9d4397398bccdb8a3e310bafc0ff706d3cb7c499R412

Reply:
right! I always forget to look and see if the GH diff has collapsed the view of a file on big changes. I see it now.